### PR TITLE
LPS-135136

### PR DESCRIPTION
**Pull request by erickmonteiroo**
@nhpatt Could you please take a look at [this](https://github.com/liferay-usm/liferay-portal/pull/216/commits/f95e2f05dfccd86efaf8d53ecd776ffa48b8fc16) commit? We had some issues where GraphQL could not set the `Type` field of an Account because the value being passed during the `Account.Type.create` call is all upper cased, leaning to not match the values on the `enum` field and just returning `null`. Forcing the value to be lowercase on the `create` method solves the issue. Thanks a lot!

### CODE REVIEW

File: modules/apps/headless/headless-admin-user/headless-admin-user-client/src/main/java/com/liferay/headless/admin/user/client/dto/v1_0/Account.java [hunk 1]
```diff
@@ -19,6 +19,7 @@
 
 import java.io.Serializable;
 
+import java.util.Locale;
```
Comment (erickmonteiroo) [inline modules/apps/headless/headless-admin-user/headless-admin-user-client/src/main/java/com/liferay/headless/admin/user/client/dto/v1_0/Account.java:22]: @nhpatt Ignore this import.

**Comment by Sharryxue** [comment]
- The new added test AccountResourceTest > testPostAccount failed, refer [console](https://testray.liferay.com/reports/production/logs/2021-08/test-1-21/test-portal-acceptance-pullrequest(master)/8315/modules-integration-mysql57-jdk8/0/0/jenkins-console.txt.gz), would you @erickmonteiroo please double check it when you get a chance? Thanks
```
     [exec] com.liferay.headless.admin.user.resource.v1_0.test.AccountResourceTest > testPostAccount FAILED
     [exec]     java.lang.AssertionError
     [exec]     	at org.junit.Assert.fail(Assert.java:87)
     [exec]     	at org.junit.Assert.fail(Assert.java:96)
     [exec]     	at com.liferay.headless.admin.user.resource.v1_0.test.AccountResourceTest._testPostAccountInvalidType(AccountResourceTest.java:498)
     [exec]     	at com.liferay.headless.admin.user.resource.v1_0.test.AccountResourceTest.testPostAccount(AccountResourceTest.java:277)
     [exec]     	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
     [exec]     	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
     [exec]     	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
```
- The failure in [rest-builder-jdk8/0/0](https://testray.liferay.com/reports/production/logs/2021-08/test-1-21/test-portal-acceptance-pullrequest(master)/8315/rest-builder-jdk8/0/0/jenkins-console.txt.gz) seems also related.
```
     [echo] diff --git a/modules/apps/headless/headless-admin-user/headless-admin-user-client/src/main/java/com/liferay/headless/admin/user/client/dto/v1_0/Account.java b/modules/apps/headless/headless-admin-user/headless-admin-user-client/src/main/java/com/liferay/headless/admin/user/client/dto/v1_0/Account.java
     [echo] index 6b180550b621..d750b441cab6 100644
     [echo] --- a/modules/apps/headless/headless-admin-user/headless-admin-user-client/src/main/java/com/liferay/headless/admin/user/client/dto/v1_0/Account.java
     [echo] +++ b/modules/apps/headless/headless-admin-user/headless-admin-user-client/src/main/java/com/liferay/headless/admin/user/client/dto/v1_0/Account.java
     [echo] @@ -19,7 +19,6 @@ import com.liferay.headless.admin.user.client.serdes.v1_0.AccountSerDes;
     [echo]  
     [echo]  import java.io.Serializable;
     [echo]  
     [echo] -import java.util.Locale;
     [echo]  import java.util.Map;
     [echo]  import java.util.Objects;
     [echo]  
     [echo] @@ -286,7 +285,7 @@ public class Account implements Cloneable, Serializable {
     [echo]  
     [echo]  		public static Type create(String value) {
     [echo]  			for (Type type : values()) {
     [echo] -				if (Objects.equals(type.getValue().toLowerCase(), value.toLowerCase())) {
     [echo] +				if (Objects.equals(type.getValue(), value)) {
     [echo]  					return type;
     [echo]  				}
     [echo]  			}
```

**Comment by nhpatt** [comment]
```if (Objects.equals(type.getValue().toLowerCase(), value.toLowerCase())) {```

I guess that you have an enumeration like this right:

enum {
    VALUE("value"), OTHER-VALUE("other value")
}

GraphQL *always* returns the enum (VALUE) because the standards says so and you are sending the internal string (value). If you add this line both will work. But OTHER-VALUE won't work.

The test is failing because that code is generated, so we should change the generation.

PD: please ping in slack when sending a PR instead of a mention, I get too many mentions :(

**Comment by nhpatt** [comment]
Sorry I just answered Drew, should be fixed when this PR: https://github.com/liferay-frontend/liferay-portal/pull/1368 passes.

**Comment by erickmonteiroo** [comment]
> Sorry I just answered Drew, should be fixed when this PR: [liferay-frontend#1368](https://github.com/liferay-frontend/liferay-portal/pull/1368) passes.

@nhpatt Thankss! 😄

**Pull request closed without merging**